lms/services/moodle.py

```diff
@@ -35,6 +35,9 @@ def group_set_groups(self, group_set_id: int) -> list[dict]:
         return [
             {"id": g["id"], "name": g["name"], "group_set_id": group_set_id}
             for g in groups
+            # While the endpoint is not course based we filter by group on our
+            # end to avoid using groups from one course in another.
+            if g["courseid"] == course_id
```

Comment:
This is important because otherwise we won't know course copy happen. We will point the new course to the old group sets.

With this we will only find groups in the course we care about.